pkg/costmodel/allocation_helpers.go

```diff
@@ -0,0 +1,133 @@
+		end := NewIntervalPoint(*podInterval.End(), "end", podKey)
+
+		intervals = append(intervals, []IntervalPoint{start, end}...)
+
+	}
+
+	sortIntervalPoints(intervals)
+
+	return intervals
+
+}
+
+// sortIntervalPoints sorts a list of IntervalPoints from earliest
+// to latest IntervalPoint.Time. In the case that two IntervalPoints
+// have the same time, the point with Type "start" is treated as coming
+// before the "end".
+func sortIntervalPoints(intervals []IntervalPoint) {
+	sort.Slice(intervals, func(i, j int) bool {
+		if intervals[i].Time.Equal(intervals[j].Time) {
+			return intervals[i].PointType == "start" && intervals[j].PointType == "end"
+		}
+		return intervals[i].Time.Before(intervals[j].Time)
+	})
+}
+
+// getPVCCostCoefficients gets a coefficient which represents the scale
+// factor that each PVC in a pvcIntervalMap and corresponding slice of
+// IntervalPoints intervals uses to calculate a cost for that PVC's PV.
+func getPVCCostCoefficients(intervals []IntervalPoint, pvcIntervalMap map[podKey]kubecost.Window, pvcCostCoefficientMap map[podKey][][]float64) {
+
+	// pvcCostCoefficientMap is mutated in this function. The format is
+	// such that the individual coefficient components are preserved for
+	// testing purposes.
+
+	var activePods float64
+
+	activeKeys := make(map[podKey]struct{})
+
+	// For each interval i.e. for any time a pod-PVC relation ends or starts...
+	for i := 1; i < len(intervals); i++ {
+
+		point := intervals[i]
+		prevPoint := intervals[i-1]
+
+		// intervals will always have at least two IntervalPoints (one start/end)
+		if i == 1 {
+			activePods += 1
+			activeKeys[prevPoint.Key] = struct{}{}
+		}
+
+		// If the current point happens at a later time than the previous point
+		if !point.Time.Equal(prevPoint.Time) {
+			for key := range activeKeys {
+				// Add a two-long slice of float64 representing:
+				// 1. The division of cost based on how many pods were running between those points
+				// 2. The ratio of the time between those points to the total time that pod was running
+				pvcCostCoefficientMap[key] = append(
+					pvcCostCoefficientMap[key],
+					[]float64{1.0 / activePods, point.Time.Sub(prevPoint.Time).Minutes() / pvcIntervalMap[key].Duration().Minutes()},
+				)
+			}
+		}
+
+		// If the point was a start, increment and track
+		if point.PointType == "start" {
+			activePods += 1
+			activeKeys[point.Key] = struct{}{}
+		}
+
+		// If the point was an end, decrement and stop tracking
+		if point.PointType == "end" {
+			activePods -= 1
+			delete(activeKeys, point.Key)
+		}
+
+	}
+}
+
+// getCoefficient takes the components of a PVC-pod PV cost coefficient
+// determined by getPVCCostCoefficient and gets the resulting single
+// floating point coefficient.
+func getCoefficient(coefficientComponents [][]float64) float64 {
+
+	coefficient := 0.0
+
+	for i := range coefficientComponents {
+
+		cost := coefficientComponents[i][0]
```

Comment:
At this point, do we expect this value (`coefficientComponents[i][0]`) to be the cost? Or is it the percentage of all pods connected to the PV? Trying to figure out if I missed where cost might have been factored into these `coefficientComponents`.

Reply:
Also -- not a huge deal, but `getCoefficient` is a relatively vague name for `package costmodel`. We might want to come up with a name more specific to this precise use-case.

Reply:
Yeah, I'll see if I can clarify both the name of this function and (by adding a struct) the coefficient component items.

I really should not have named that cost. You are correct that it is the percentage of all pods connected to the PV. Totally my bad, I think I was thinking cost as in "proportion of cost".